src/audio/google/google_ctc_audio_processing.c

```diff
@@ -0,0 +1,359 @@
+
+	if (!comp_is_current_data_blob_valid(cd->tuning_handler) &&
+	    !comp_is_new_data_blob_available(cd->tuning_handler)) {
+		/*
+		 * The data blob hasn't been available once so far.
+		 *
+		 * This looks redundant since the same check will be done in
+		 * comp_get_data_blob() below. But without this early return,
+		 * hundreds of warn message lines are produced per second by
+		 * comp_get_data_blob() calls until the data blob is arrived.
+		 */
+		return 0;
+	}
+
+	config = comp_get_data_blob(cd->tuning_handler, &size, NULL);
+	if (size == 0) {
+		/* No data to be handled */
+		return 0;
+	}
+
+	if (!config) {
+		comp_err(dev, "google_ctc_audio_processing_reconfigure(): Tuning config not set");
+		return -EINVAL;
+	}
+
+	comp_info(dev, "google_ctc_audio_processing_reconfigure(): New tuning config %p (%zu bytes)",
+		  config, size);
+
+	cd->reconfigure = false;
+
+	uint8_t *google_ctc_audio_processing_config;
+	size_t google_ctc_audio_processing_config_size;
+	bool google_ctc_audio_processing_config_present;
+
+	GoogleCtcAudioProcessingParseSofConfigMessage(config, size,
+						      &google_ctc_audio_processing_config,
+						      &google_ctc_audio_processing_config_size,
+						      &google_ctc_audio_processing_config_present);
+
+	if (google_ctc_audio_processing_config_present) {
+		comp_info(dev,
+			  "google_ctc_audio_processing_reconfigure(): Applying config of size %zu bytes",
+			  google_ctc_audio_processing_config_size);
+
+		ret = GoogleCtcAudioProcessingReconfigure(cd->state,
+							  google_ctc_audio_processing_config,
+							  google_ctc_audio_processing_config_size);
+		if (ret) {
+			comp_err(dev, "GoogleCtcAudioProcessingReconfigure failed: %d",
+				 ret);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+static int ctc_prepare(struct processing_module *mod)
+{
+	comp_info(mod->dev, "ctc_prepare()");
+
+	return 0;
+}
+
+static int ctc_reset(struct processing_module *mod)
+{
+	comp_dbg(mod->dev, "ctc_reset()");
+
+	return 0;
+}
+
+static int ctc_process(struct processing_module *mod,
+					       struct sof_source **sources, int num_of_sources,
+					       struct sof_sink **sinks, int num_of_sinks)
+{
+	struct google_ctc_audio_processing_comp_data *cd = module_get_private_data(mod);
+
+	int16_t const *in;
+	int8_t const *in_buf_start;
+	int8_t const *in_buf_end;
+	size_t in_buf_size;
+
+	int16_t *out;
+	int8_t *out_buf_start;
+	int8_t *out_buf_end;
+	size_t out_buf_size;
+
+	size_t num_of_bytes_to_process;
+
+	struct sof_source *in_stream;
+	struct sof_sink *out_stream;
+
+	comp_dbg(dev, "ctc_process()");
+
+	if (cd->reconfigure) {
+		ret = google_ctc_audio_processing_reconfigure(mod);
+		if (ret)
+			return ret;
+	}
+
+	in_stream = sources[0];
+	out_stream = sinks[0];
+
+	num_of_bytes_to_process = cd->num_frames * source_get_frame_bytes(in_stream);
+
+	ret = source_get_data(in_stream, num_of_bytes_to_process, (const void **)&in,
+			      (const void **)&in_buf_start,  &in_buf_size);
+	assert(!ret);
```

Comment:
I wouldn't crash the system in such cases, just return the error code. Same below

Reply:
I changed the process function to avoid that. Thanks!